lib/helpers.php

```diff
@@ -149,7 +153,7 @@ function micropub_post($endpoint, $params, $access_token, $file_path = NULL, $js
     $mimetype = finfo_file($finfo, $file_path);
     $multipart = new p3k\Multipart();
     $multipart->addArray($params);
-    $multipart->addFile($file_prop, $file_path, $mimetype);
+    $multipart->addFile($file_prop, $filename, $mimetype, $file_content);
```

Comment:
The function `addFile` in `p3k\Multipart` does not take 4 arguments. Did you modify that function as well? If so, you can send a PR to that project for that as well. https://github.com/aaronpk/php-multipart-encoder